Pinta.Tools/Tools/EraserTool.cs

```diff
@@ -32,98 +32,244 @@
 
 namespace Pinta.Tools
 {
-	public class EraserTool : BaseBrushTool
-	{		
-		private Point last_point = point_empty;
-
-		public EraserTool ()
-		{
-		}
-
-		protected override void OnBuildToolBar(Toolbar tb)
-		{
-			base.OnBuildToolBar(tb);
-
-			// Change the cursor when the BrushWidth is changed.
-			brush_width.ComboBox.Changed += (sender, e) => SetCursor (DefaultCursor);
-		}
-
-		#region Properties
-		public override string Name { get { return Catalog.GetString ("Eraser"); } }
-		public override string Icon { get { return "Tools.Eraser.png"; } }
-		public override string StatusBarText { get { return Catalog.GetString ("Left click to erase to transparent, right click to erase to secondary color. "); } }
-
-		public override Gdk.Cursor DefaultCursor {
-			get {
-				int iconOffsetX, iconOffsetY;
-				var icon = CreateIconWithShape ("Cursor.Eraser.png",
-				                                CursorShape.Ellipse, BrushWidth, 8, 22,
-				                                out iconOffsetX, out iconOffsetY);
+    public class EraserTool : BaseBrushTool
+    {       
+        protected enum EraserType
+        {
+            Normal = 0,
+            Smooth = 1,
+        }
+
+        protected Point last_point = point_empty;
+        protected EraserType eraser_type = EraserType.Normal; 
+
+        protected const int LUT_Resolution = 256;
+        protected byte[][] lut_factor = null;
+        protected ImageSurface tmp_surface = null;
```

Comment:
`ImageSurface` implements `IDisposable`, so I think there are some missing `Dispose()` calls. I think it would be better to return the surface from `copySurfacePart` and have a `using` statement inside `eraseSmooth` so that the lifetime is more explicit.